evidence/pool.go

```diff
@@ -218,7 +263,59 @@ func (evpool *Pool) ValidatorLastHeight(address []byte) int64 {
 	return h
 }
 
-func (evpool *Pool) removeEvidence(
+// State returns the current state of the evpool.
+func (evpool *Pool) State() sm.State {
+	evpool.mtx.Lock()
+	defer evpool.mtx.Unlock()
+	return evpool.state
+}
+
+func (evpool *Pool) addPendingEvidence(evidence types.Evidence) error {
+	var err error
+	evBytes := cdc.MustMarshalBinaryBare(evidence)
+	key := keyPending(evidence)
+	if err = evpool.evidenceStore.Set(key, evBytes); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (evpool *Pool) removePendingEvidence(evidence types.Evidence) {
+	key := keyPending(evidence)
+	if err := evpool.evidenceStore.Delete(key); err != nil {
+		evpool.logger.Error("Unable to delete pending evidence", "err", err)
+	}
+}
+
+// listEvidence lists up to maxNum pieces of evidence for the given prefix key.
+// It is wrapped by PriorityEvidence and PendingEvidence for convenience.
+// If maxNum is -1, there's no cap on the size of returned evidence.
+func (evpool *Pool) listEvidence(prefixKey byte, maxNum int64) (evidence []types.Evidence) {
+	var count int64
+	iter, err := dbm.IteratePrefix(evpool.evidenceStore, []byte{prefixKey})
+	if err != nil {
+		panic(err)
```

Comment:
does it make sense to return error instead of panicking?

Reply:
Yes I would agree